src/core/services/compiler/compiler.js

```diff
@@ -444,27 +454,33 @@ function MdCompilerProvider($compileProvider) {
 
   /**
    * Creates and instantiates a new controller with the specified options.
-   * @param {!Object} options Options that include the controller
+   * @param {!Object} options Options that include the controller function or string.
    * @param {!Object} injectLocals Locals to to be provided in the controller DI.
    * @param {!Object} locals Locals to be injected to the controller.
    * @returns {!Object} Created controller instance.
    */
   MdCompilerService.prototype._createController = function(options, injectLocals, locals) {
-    // The third and fourth arguments to $controller are considered private and are undocumented:
-    // https://github.com/angular/angular.js/blob/master/src/ng/controller.js#L86
+    var ctrl;
+    // The third argument to $controller is considered private and undocumented:
+    // https://github.com/angular/angular.js/blob/v1.6.10/src/ng/controller.js#L102-L109.
+    // TODO remove the use of this third argument in AngularJS Material 1.2.0.
     // Passing `true` as the third argument causes `$controller` to return a function that
-    // gets the controller instance instead returning of the instance directly. When the
+    // gets the controller instance instead of returning the instance directly. When the
     // controller is defined as a function, `invokeCtrl.instance` is the *same instance* as
-    // `invokeCtrl()`. However, then the controller is an ES6 class, `invokeCtrl.instance` is a
+    // `invokeCtrl()`. However, when the controller is an ES6 class, `invokeCtrl.instance` is a
     // *different instance* from `invokeCtrl()`.
-    var invokeCtrl = this.$controller(options.controller, injectLocals, true, options.controllerAs);
+    var invokeCtrl = this.$controller(options.controller, injectLocals, true);
```

Comment:
Could you switch the logic to not pass the third parameter if `getPreAssignBindingsEnabled() === false`? This would be safer as in the recommended usage no private API would be used.

Reply:
Good call. I think it makes the 1.2.0 update more straightforward as well.

Reply:
PTAL and let me know if I missed anything.